soweego/validator/checks.py

```diff
@@ -540,27 +582,25 @@ def links(
       of URL domains. Default: ``False``
     :param wd_cache: (optional) a ``dict`` of links gathered from Wikidata
       in a previous run. Default: ``None``
-    :return: ``tuple`` of 6 objects
+    :return: 7 objects
 
       1. ``dict`` of identifiers that should be deprecated
       2. ``list`` of third-party identifiers that should be added
       3. ``list`` of URLs that should be added
       4. ``list`` of third-party identifiers that should be referenced
       5. ``list`` of URLs that should be referenced
-      6. ``dict`` of links gathered from Wikidata
+      6. ``list`` of URLs found in Wikidata but not in the target catalog
+      7. ``dict`` of links gathered from Wikidata
 
     """
     # Target catalog side first:
     # enable early return in case of no target links
     target_links = data_gathering.gather_target_links(entity, catalog)
     if target_links is None:
-        return None, None, None, None, None, None
+        return None, None, None, None, None, None, None
```

Comment:
The declared return type does not match what you return here, so you need to change either one of them:

 - The return type it should be `Tuple[Optional[defaultdict], Optional[list], (etc)]]`
 - Or change this to `return Non`

PS this is caught by [mypy](http://mypy-lang.org/), aren't you using it?
```
checks.py:600: error: Incompatible return value type (got "Tuple[None, None, None, None, None, None, None]", expected "Optional[Tuple[defaultdict[Any, Any], List[Any], List[Any], List[Any], List[Any], List[Any], Dict[Any, Any]]]")
```

Reply:
> PS this is caught by mypy, aren't you using it?

To cut a long story short: when we started the project, type hints were relatively new. We decided to add them just to relevant public functions, hence no automatic check was implemented.
Later, we put a script in place, and we should definitely automate its execution: https://github.com/Wikidata/soweego/blob/9ed8b4e28fd2ea4150b37cbb49b44ce8bfb2c381/scripts/enforce_style.sh